website/addons/s3/static/user-cfg.js

```diff
@@ -1,61 +1,10 @@
-'use strict';
+var S3ViewModel = require('./s3UserConfig.js').S3ViewModel;
+var S3UserConfig = require('./s3UserConfig.js').S3UserConfig;
 
-var $ = require('jquery');
-var bootbox = require('bootbox');
-var AddonHelper = require('js/addonHelper');
-var language = require('js/osfLanguage').Addons.s3;
+// Endpoint for S3 user settings
+var url = '/api/v1/settings/s3/accounts/';
 
-$('#s3RemoveAccess').on('click', function() {
-    bootbox.confirm({
-        title: 'Disconnect S3 Account?',
-        message: language.confirmDeauth,
-        callback: function(result) {
-            if(result) {
-                deleteToken();
-            }
-        },
-        buttons:{
-            confirm:{
-                label:'Disconnect',
-                className:'btn-danger'
-            }
-        }
-    });
-});
+var s3UserConfig = new S3UserConfig('#s3AddonScope', url);
 
-function deleteToken() {
-    var $this = $(this),
-    addon = $this.attr('data-addon'),
-    msgElm = $this.find('.addon-settings-message');
-    $.ajax({
-        type: 'DELETE',
-        url: '/api/v1/settings/s3/',
-        contentType: 'application/json',
-        dataType: 'json',
-        success: function(response) {
-            msgElm.text('Keys removed')
-                .removeClass('text-danger').addClass('text-success')
-                .fadeOut(100).fadeIn();
-            window.location.reload();
-        },
-        error: function(xhr) {
-            var response = JSON.parse(xhr.responseText);
-            if (response && response.message) {
-                if(response.message === 'reload')
-                    window.location.reload();
-                else
-                    message = response.message;
-            } else {
-                message = 'Error: Keys not removed';
-            }
-            msgElm.text(message)
-                .removeClass('text-success').addClass('text-danger')
-                .fadeOut(100).fadeIn();
-        }
-    });
-    return false;
-}
-
-$(document).ready(function() {
-    $(window.contextVars.addonSettingsSelector).on('submit', AddonHelper.onSubmitSettings);
-});
+// Load initial S3 data
+s3UserConfig.viewModel.updateAccounts();
```

Comment:
Since this happens when `s3UserConfig.viewModel` is initialized, maybe move it to the `S3ViewModel` constructor?